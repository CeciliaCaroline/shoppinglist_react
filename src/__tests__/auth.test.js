import React from 'react';
import expect from 'expect';
import renderer from 'react-test-renderer';
import Register from "../components/auth/register";
import Login from "../components/auth/login";
import SendEmail from "../components/auth/email_send";
import ResetPassword from "../components/auth/reset_password";
import Header from "../components/header";
import "../setup_test";


describe('Authentication components', () => {

    it('renders correctly', () => {
        const register = renderer.create(<Register/>).toJSON();
        expect(register).toMatchSnapshot();
    });

    it('renders correctly', () => {
        const login = renderer.create(<Login/>).toJSON();
        expect(login).toMatchSnapshot();

    });
    it('renders correctly', () => {
        const sendEmail = renderer.create(<SendEmail/>).toJSON();
        expect(sendEmail).toMatchSnapshot();

    });
    it('renders correctly', () => {
        const reset = renderer.create(<ResetPassword/>).toJSON();
        expect(reset).toMatchSnapshot();
    });
    it('renders correctly', () => {
        const header = renderer.create(<Header/>).toJSON();
        expect(header).toMatchSnapshot();
    });
});

