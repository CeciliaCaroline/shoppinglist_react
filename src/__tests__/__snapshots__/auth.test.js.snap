// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication components renders correctly 1`] = `
<div
  className="container mt-5"
>
  <div
    className="row"
  >
    <div
      className="notifications-wrapper"
      style={Object {}}
    />
    <div
      className="container form-background card mx-5 mt-5 col media card "
    >
      <h2
        className="mt-5 text-capitalize"
      >
         Shopping list
      </h2>
      <h6
        className="text-capitalize"
      >
        The Best Way To Record and Share your shopping lists
      </h6>
      <p
        className="lead"
      >
        This is an application that enables users to record, track and share things they want to spend money on in order to meet their desires.
      </p>
      <h4>
        Create
      </h4>
      <p>
        Create a shopping list with all the items that you want to purchase
      </p>
      <h4>
        Update
      </h4>
      <p>
        Change your mind? Update your list with new items
      </p>
      <h4>
        Share
      </h4>
      <p>
        Share your list with friends and family
      </p>
    </div>
    <form
      className="container media-body form-background card mx-5 mt-5 col"
      onSubmit={[Function]}
    >
      <h2
        className="text-capitalize card-title mt-5"
      >
        New to Shoppinglist. Register
      </h2>
      <div
        className="form-group"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="form-control"
          name="email"
          onChange={[Function]}
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="password"
        >
          Confirm Password
        </label>
        <input
          className="form-control"
          name="confirm_password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
      <div>
        <button
          className="btn btn-primary btn-block"
          onClick={[Function]}
          type="submit"
        >
          Sign up
        </button>
        <p
          className="text-center"
        >
          Already have an account??
          <a
            href="/auth/login"
            onClick={[Function]}
          >
            Login
          </a>
        </p>
      </div>
    </form>
  </div>
</div>
`;

exports[`Authentication components renders correctly 2`] = `
<div
  className="container items"
>
  <form
    className="container items form-background card mt-5 col-6 "
    onSubmit={[Function]}
  >
    <div
      className="notifications-wrapper"
      style={Object {}}
    />
    <h2
      className="text-center"
    >
      Log In
    </h2>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="form-control"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="text-center"
    >
      <button
        className="btn btn-primary btn-block"
        type="submit"
      >
        Log In
      </button>
      <small
        className="text-center"
      >
        Forgot Password? You can reset it 
        <a
          href="/auth/reset_password"
          onClick={[Function]}
        >
          here
        </a>
      </small>
      <div
        className="text-center"
      >
        <small
          className="text-center justify content-end"
        >
          <a
            href="/auth/register"
            onClick={[Function]}
          >
             Sign Up
          </a>
        </small>
      </div>
    </div>
  </form>
</div>
`;

exports[`Authentication components renders correctly 3`] = `
<div
  className="container items"
>
  <div
    className="notifications-wrapper"
    style={Object {}}
  />
  <form
    className="container items form-background card mt-5 col-6 "
    onSubmit={[Function]}
  >
    <h2
      className="text-center"
    >
      Reset Password
    </h2>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="form-control"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
    </div>
    <div>
      <button
        className="btn btn-primary btn-block"
        type="submit"
      >
        Reset Password
      </button>
    </div>
  </form>
</div>
`;

exports[`Authentication components renders correctly 4`] = `
<div
  className="container items"
>
  <div
    className="notifications-wrapper"
    style={Object {}}
  />
  <form
    className="container items form-background card mt-5 col-6 "
    onSubmit={[Function]}
  >
    <h2
      className="text-center"
    >
      Reset Password
    </h2>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="form-control"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        New Password
      </label>
      <input
        className="form-control"
        name="new_password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Confirm Password
      </label>
      <input
        className="form-control"
        name="confirm_password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div>
      <button
        className="btn btn-primary btn-block"
        type="submit"
      >
        Reset Password
      </button>
    </div>
  </form>
</div>
`;

exports[`Authentication components renders correctly 5`] = `
<nav
  className="navbar navbar-inverse fixed-top bg-faded navbar-toggleable-sm "
>
  <div
    className="container"
  >
    <a
      className="navbar-brand m-o align-content-end mr-auto"
      href="/v2/shoppinglist/"
      onClick={undefined}
    >
      SHOPPING LIST
    </a>
    <div
      className="navbar-nav "
    >
      <a
        className="nav-item nav-link active"
        href="/v2/shoppinglist/"
        onClick={undefined}
      >
        My Lists
      </a>
      <a
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Log Out
      </a>
    </div>
  </div>
</nav>
`;
